### PR TITLE
Tab completion now takes into account all worlds.

### LINKED ISSUE #2563: Playername tab completion issue
Tab completion for playernames doesn't work if the player is located in another world than you are.

### PR DESCRIPTION
**Pull request by bearbin**
Should fix #2563

### CODE REVIEW

**Comment by bearbin** [comment]
This code is wrong, will fix when I get home.

**Comment by NiLSPACE** [comment]
We could make this configurable. Some people might want their world's separated.

**Comment by bearbin** [comment]
I think configurability is a good idea. Would the code so far work for the basic case?

**Comment by NiLSPACE** [comment]
Yes, I think so, but you/your editor removed allot of tabs. Is that a problem for anyone?

**Comment by SphinxC0re** [comment]
That's fine

**Comment by Seadragon91** [comment]
Well I think @madmaxoft doesn't like that. Makes diff unreadable.

File: src/ClientHandle.cpp [hunk 1]
```diff
@@ -3073,7 +3093,3 @@ void cClientHandle::OnError(int a_ErrorCode, const AString & a_ErrorMsg)
 	}
 	SocketClosed();
 }
-
-
-
-
```
Comment (SphinxC0re) [inline src/ClientHandle.cpp]: You should keep these 4 lines

File: src/ClientHandle.cpp [hunk 2]
```diff
@@ -1788,7 +1788,27 @@ void cClientHandle::HandleUnmount(void)
 void cClientHandle::HandleTabCompletion(const AString & a_Text)
 {
 	AStringVector Results;
-	m_Player->GetWorld()->TabCompleteUserName(a_Text, Results);
+
+	class cWorldCallback : public cWorldListCallback
+	{
+	public:
+		cWorldCallback(AStringVector results, const AString & text) :
```
Comment (tigerw) [inline src/ClientHandle.cpp]: Naming convention remix! :P

File: src/ClientHandle.cpp [hunk 3]
```diff
@@ -1788,7 +1788,27 @@ void cClientHandle::HandleUnmount(void)
...
+		{
+		}
+
+		virtual bool Item(cWorld * a_World) override
+		{
+			a_World->TabCompleteUserName(m_Text, Results);
+			return true;
+		}
+	private:
+		AStringVector Results;
```
Comment (tigerw) [inline src/ClientHandle.cpp]: Should this not be a reference?

File: src/ClientHandle.cpp (hunk 2 above)
Comment (SphinxC0re) [inline src/ClientHandle.cpp]: :musical_note:  *scratch sounds*

**Comment by SphinxC0re** [comment]
What about configurability?

**Comment by bearbin** [comment]
I'm still working on it, it should be done tomorrow.

**Comment by mathias-github** [comment]
Any updates on this?

**Comment by bearbin** [comment]
I haven't abandoned the PR, I just haven't had the time to work on it. I should be able to do a bit this weekend.

**Comment by SphinxC0re** [comment]
:stuck_out_tongue_winking_eye: @bearbin by "tomorrow" you meant next year?

**Comment by bearbin** [comment]
Sorry...

You can close this if you want, I will probably work on it "soon".

**Comment by SphinxC0re** [comment]
I won't since you said you'll work on it

**Comment by LogicParrot** [comment]
How about merging the basic case and doing configurability later?

**Pull request closed without merging**